python/ray/autoscaler/v2/instance_manager/reconciler.py

```diff
@@ -0,0 +1,226 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+from ray.autoscaler.v2.instance_manager.node_provider import (
+    CloudInstance,
+    CloudInstanceId,
+    CloudInstanceProviderError,
+    ICloudInstanceProvider,
+)
+from ray.core.generated.autoscaler_pb2 import ClusterResourceState
+from ray.core.generated.instance_manager_pb2 import Instance as IMInstance
+from ray.core.generated.instance_manager_pb2 import (
+    InstanceUpdateEvent as IMInstanceUpdateEvent,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class IReconciler(ABC):
```

Comment:
yeah, i agree, the interface doesn't add much impl value here. I would like to keep it since it does allow us to better organize the various kinds of reconcilers.